common/plugins/org.jboss.tools.common.ui/src/org/jboss/tools/common/ui/JobResultFuture.java

```diff
@@ -41,12 +41,13 @@ public JobResultFuture(Job job) {
 
 	@Override
 	public boolean cancel(boolean mayInterruptIfRunning) {
-		if ((isRunning(job)
-				&& mayInterruptIfRunning)
-				|| !isRunning(job)) {
```

Comment:
It may be still waiting or sleeping. JobManager does all the checks and changes state to NONE. So, even if the current state of job is NONE, it does not harm to call cancel().
For the boolean expression, I only made it a bit shorter, but it is equivalent to what it was.